src/main/java/com/hubspot/jinjava/lib/expression/EagerExpressionStrategy.java

```diff
@@ -11,6 +21,113 @@ public RenderedOutputNode interpretOutput(
+    EagerStringResult eagerStringResult = eagerResolveExpression(master, interpreter);
+    return new RenderedOutputNode(
+      eagerStringResult.getPrefixToPreserveState() + eagerStringResult.getResult()
+    );
+  }
+
+  private EagerStringResult eagerResolveExpression(
+    ExpressionToken master,
+    JinjavaInterpreter interpreter
+  ) {
+    ChunkResolver chunkResolver = new ChunkResolver(
+      master.getExpr(),
+      master,
+      interpreter
+    );
+    EagerStringResult resolvedExpression = EagerTagDecorator.executeInChildContext(
+      eagerInterpreter -> chunkResolver.resolveChunks(),
+      interpreter,
+      true
+    );
+    StringBuilder prefixToPreserveState = new StringBuilder(
+      interpreter.getContext().isProtectedMode()
+        ? resolvedExpression.getPrefixToPreserveState()
+        : ""
+    );
+    if (chunkResolver.getDeferredWords().isEmpty()) {
+      String result = WhitespaceUtils.unquote(resolvedExpression.getResult());
+      if (
+        !StringUtils.equals(result, master.getImage()) &&
+        (
+          StringUtils.contains(result, master.getSymbols().getExpressionStart()) ||
+          StringUtils.contains(result, master.getSymbols().getExpressionStartWithTag())
+        )
+      ) {
+        if (interpreter.getConfig().isNestedInterpretationEnabled()) {
+          try {
+            result = interpreter.renderFlat(result);
+          } catch (Exception e) {
+            Logging.ENGINE_LOG.warn("Error rendering variable node result", e);
+          }
+        } else {
+          // Possible macro/set tag in front of this one. Includes result
+          result = wrapInRawOrExpressionIfNeeded(result, interpreter);
+        }
+      }
+
+      if (interpreter.getContext().isAutoEscape()) {
+        result = EscapeFilter.escapeHtmlEntities(result);
+      }
+      return new EagerStringResult(result, prefixToPreserveState.toString());
+    }
+    prefixToPreserveState.append(
+      EagerTagDecorator.reconstructFromContextBeforeDeferring(
+        chunkResolver.getDeferredWords(),
+        interpreter
+      )
+    );
+    String helpers = wrapInExpression(resolvedExpression.getResult(), interpreter);
+    interpreter
+      .getContext()
+      .handleEagerToken(
+        new EagerToken(
+          new TagToken(
+            helpers,
+            master.getLineNumber(),
+            master.getStartPosition(),
+            master.getSymbols()
+          ),
+          chunkResolver.getDeferredWords()
+        )
+      );
+    // There is no result because it couldn't be entirely evaluated.
+    return new EagerStringResult(
+      "",
+      EagerTagDecorator.wrapInAutoEscapeIfNeeded(
+        prefixToPreserveState.toString() + helpers,
+        interpreter
+      )
+    );
+  }
+
+  private static String wrapInRawOrExpressionIfNeeded(
+    String output,
+    JinjavaInterpreter interpreter
+  ) {
+    if (
+      interpreter.getConfig().getExecutionMode().isPreserveRawTags() &&
```

Comment:
Might want to a create a variable for `interpreter.getConfig()` to shorten up these lines.